### PR TITLE
[stable] Prepare 1.62.0 stable release

### PR DESCRIPTION
**Pull request by emilyalbini**
This PR backports the latest release note changes from master, and prepares the stable artifacts.

r? @ghost

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,254 @@
...
+- [Implement a lint to warn about unused macro rules][96150]
+- [Promote `x86_64-unknown-none` target to Tier 2\*][95705]
+
+\* Refer to Rust's [platform support page][platform-support-doc] for more
+   information on Rust's tiered platform support.
+
+Libraries
+---------
+
+- [Move `CStr` to libcore, and `CString` to liballoc][94079]
```
Comment (wwylele) [inline RELEASES.md:32]: From my understanding of https://github.com/rust-lang/rust/issues/98314, this is still unstable. Should this be in the stable release note?

Comment (Mark-Simulacrum) [reply RELEASES.md:32]: Seems like an oversight. If you'd like to file a PR against the master branch, that'd be great!

Comment (wwylele) [reply RELEASES.md:32]: I opened https://github.com/rust-lang/rust/pull/98597

**Pull request merged**